File: MSMA_Backend/src/main/java/com/spring/dto/response/SongResponse.java
package com.spring.dto.response;

import lombok.*;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder(toBuilder = true)
public class SongResponse {
    private Long id;
    private String title;
    private String releaseDate;
    private String lyrics;
    private String duration;
    private String imageUrl;
    private Boolean downloadPermission;
    private String description;
    private String mp3Url;
    private String songStatus;
    private List<String> genreNameList;
    private List<String> artistNameList;
    private Long numberOfListeners;
    private Long countListen;
    private Long numberOfUserLike;
    private Long numberOfDownload;
}

File: MSMA/app/src/main/java/vn/edu/usth/msma/data/dto/response/management/SongResponse.kt
package vn.edu.usth.msma.data.dto.response.management

import com.google.gson.annotations.SerializedName

data class SongResponse(
    @SerializedName("id")
    val id: Long,

    @SerializedName("title")
    val title: String,

    @SerializedName("releaseDate")
    val releaseDate: String,

    @SerializedName("lyrics")
    val lyrics: String,

    @SerializedName("duration")
    val duration: String,

    @SerializedName("imageUrl")
    val imageUrl: String,

    @SerializedName("downloadPermission")
    val downloadPermission: Boolean,

    @SerializedName("description")
    val description: String,

    @SerializedName("mp3Url")
    val mp3Url: String,

    @SerializedName("songStatus")
    val songStatus: String,

    @SerializedName("genreNameList")
    val genreNameList: List<String>,

    @SerializedName("artistNameList")
    val artistNameList: List<String>,

    @SerializedName("numberOfListeners")
    val numberOfListeners: Long,

    @SerializedName("countListen")
    val countListen: Long,

    @SerializedName("numberOfUserLike")
    val numberOfUserLike: Long,

    @SerializedName("numberOfDownload")
    val numberOfDownload: Long
)
